examples/SimpleRagas/SimpleRagas.ipynb

```diff
@@ -0,0 +1,1018 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "braintrust.login(\n",
+    "    api_key=os.environ.get(\"BRAINTRUST_API_KEY\", \"Your BRAINTRUST_API_KEY here\")\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "q_and_a_dict = {d[\"input\"]: d[\"answer\"] for d in rag_results_w_context}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Experiment gpt-3.5-turbo picks 2 out of 5, answer w gpt-3.5-turbo-cdd11503 is running at https://www.braintrust.dev/app/braintrustdata.com/p/Ragas%20Retrieval/experiments/gpt-3.5-turbo%20picks%202%20out%20of%205%2C%20answer%20w%20gpt-3.5-turbo-cdd11503\n",
+      "Ragas Retrieval [experiment_name=gpt-3.5-turbo picks 2 out of 5, answer w gpt-3.5-turbo] (data): 20it [00:00, 24328.91it/s]\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "10589aa1de15496ba8c2c5719e81cdfe",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Ragas Retrieval [experiment_name=gpt-3.5-turbo picks 2 out of 5, answer w gpt-3.5-turbo] (tasks):   0%|       …"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "=========================SUMMARY=========================\n",
+      "gpt-3.5-turbo picks 2 out of 5, answer w gpt-3.5-turbo-cdd11503 compared to No Document Selection, Topk=5, model=gpt-4o-4bbf0252:\n",
+      "88.73% 'ContextRecall'       score\n",
+      "66.46% (-00.90%) 'AnswerCorrectness'   score\t(9 improvements, 11 regressions)\n",
+      "100.00% (-) 'Faithfulness'        score\t(0 improvements, 0 regressions)\n",
+      "52.10% (+28.46%) 'ContextRelevancy'    score\t(18 improvements, 2 regressions)\n",
+      "58.57% (+07.15%) 'ContextEntityRecall' score\t(10 improvements, 10 regressions)\n",
+      "\n",
+      "5.98s (+353.49%) 'duration'\t(0 improvements, 20 regressions)\n",
+      "\n",
+      "See results for gpt-3.5-turbo picks 2 out of 5, answer w gpt-3.5-turbo-cdd11503 at https://www.braintrust.dev/app/braintrustdata.com/p/Ragas%20Retrieval/experiments/gpt-3.5-turbo%20picks%202%20out%20of%205%2C%20answer%20w%20gpt-3.5-turbo-cdd11503\n"
+     ]
+    }
+   ],
+   "source": [
+    "from autoevals import (\n",
+    "    ContextEntityRecall,\n",
+    "    ContextRelevancy,\n",
+    "    ContextRecall,\n",
+    "    Faithfulness,\n",
+    "    AnswerCorrectness,\n",
+    ")\n",
+    "from braintrust import Eval\n",
+    "\n",
+    "# TODO?: Could put some effort into finding a way to reduce this boilerplate code, which I think can be really confusing to a new braintrust user\n",
```

Comment:
I've been thinking about this too. I think if we populate the variable scope for the templating in autoevals as something like

```
{
   ...metadata,
   ...output,
   ...input,
   input,
   output,
   metadata
}
```

It would just let you use `ContextRecall`, etc. directly. WDYT?

Reply:
This would be an improvement, for sure. Sort of dovetails with my [slack message](https://braintrustdata.slack.com/archives/C074HJ5M7AP/p1716404686840899?thread_ts=1716396464.798259&cid=C074HJ5M7AP) of the current autoevals forcing us to first run RAG, and then put the results into an Eval, as opposed to what I'd like, which is wrapping the "production" (i.e. functioning) rag code in an Eval.

Reply:
Addressed this above. You can build up metadata as you run the `task`